content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+Clients can start an auth flow in one of two ways:
+
+- starting with a public account identifier, provided by the user: handle or DID
+- starting with a server hostname, provided by the user: PDS or entryway, mapping to either Resource Server and/or Authorization Server
+
+One use case for starting with a server instead of an account identifier is when the user does not remember their full account handle or only knows their account email.  Another is for authentication when a user’s handle is broken. The user will still need to know their hosting provider in these situation.
+
+When starting with an account identifier, the client must resolve the atproto identity to a DID document. If starting with a handle, it is critical (mandatory) to bidirectionally verify the handle by checking that the DID document claims the handle (see atproto Handle specification). All handle resolution techniques and all atproto-blessed DID methods must be supported to ensure interoperability with all accounts.
+
+In some client environments, it may be difficult to resolve all identity types. For example, handle resolution may involve DNS TXT queries, which are not directly supported from browser apps. Client implementations might use alternative techniques (such as DNS-over-HTTP) or could make use of a supporting web service to resolve identities.
+
+Because authorization flows are security-critical, any caching of identity resolution should choose cache lifetimes carefully. Cache lifetimes of less than 10 minutes are recommended for auth flows specifically.
+
+The resolved DID should be bound to the overall auth session and should be used as the primary account identifier within client app code. Handles (when verified) are acceptable to display in user interfaces, but may change over time and need to be re-verified periodically. When passing an account identifier through to the Authorization Server as part of the Authorization Request in the `login_field`, it is recommended to use the exact account identifier supplied by the user (handle or DID) to ensure any sign-in flow is consistent (users might not recognize their own account DID).
+
+At the end of the auth flow, when the client does an initial token fetch, the Authorization Server must return the account DID in the `sub` field of the JSON response body. If the entire auth flow started with an account identifier, it is critical for the client to verify that this DID matches the expected DID bound to the session earlier; the linkage from account to Authorization Server will already have been verified in this situation.
```

Comment:
Maybe mention that this mechanism is made possible though the "state" auth param

Reply:
RFC language here,

a client MUST verify instead of "it is critical for a client to"

Reply:
@matthieusieben that feels more like an implementation guide level of "how to do it" guidance?

Reply:
I believe that pointing at the `state` param could clarify, for people used to work with OIDC nonces, that in our case, the entire "identifier" for the auth request is the `state` and nothing else.

Maybe you are right that this is more of an implementation guide level but things gets so confusing when you work with oauth that I believe it might be worth making things as clear as possible as soon as possible.

Reply:
tracking here: https://github.com/bluesky-social/atproto-website/issues/339